Fix TWAP and Limit Order Status Display

fairlighteth opened the pull request:
# Fix TWAP and Limit Order Status Display

This PR improves how order statuses are displayed in the "fills at" column for both TWAP parent orders and limit orders.

## Changes

### TWAP Parent Orders
- Added status display for when all child orders are expired
- Fixed status display logic to show:
  - "Order filled" when all child orders are 100% filled
  - "Order partially filled" when all child orders have some fills
  - "Order cancelled" when all child orders are cancelled
  - "Order expired" when all child orders are expired

### Limit Orders
- Fixed status precedence in the "fills at" column and status box:
  1. CANCELLED (takes precedence as final state)
  2. EXPIRED (final state)
  3. FILLED/PARTIALLY FILLED (execution state)
- A cancelled order will now always show:
  - "Order cancelled" in both the "fills at" column and status box
  - The fill percentage bar still shows how much was filled before cancellation for historical context

## Testing
Verified display for various order states:
- TWAP parent orders with all children expired
- TWAP parent orders with mixed child states
- Limit orders cancelled after partial fills
- Limit orders with various fill percentages

## Visual Examples
- A cancelled limit order with 89.85% fills shows:
  - "Order cancelled" in both the "fills at" column and status box
  - 89.85% in the progress bar to show historical fill amount
  
  
<img width="1367" alt="Screenshot 2025-01-16 at 17 41 32" src="https://github.com/user-attachments/assets/d67a1323-05dc-49fd-8c60-1e98b91b991e" />
<img width="1372" alt="Screenshot 2025-01-16 at 17 13 22" src="https://github.com/user-attachments/assets/01d3ee64-9459-401f-8c6d-f5cac5c34a15" />
<img width="1338" alt="Screenshot 2025-01-16 at 17 11 45" src="https://github.com/user-attachments/assets/b9c921ba-4ebc-43f8-a63c-1332b5705f7e" />
<img width="1349" alt="Screenshot 2025-01-16 at 17 46 26" src="https://github.com/user-attachments/assets/7de83b85-9afb-450c-aadd-ea29aa5e3e45" />

elena-zh reviewed (changes requested):
Hey @fairlighteth , great, but there are still some issues; 
1. Partially filled TWAP parent order does not show this message in the Fills at column
![st](https://github.com/user-attachments/assets/0a7064d2-002a-41c5-a731-2e0104c5a9d4)
2. No status is displayed for a parent TWAP order when it is cancelled+expired: 
![image](https://github.com/user-attachments/assets/f64dc698-8e17-46b5-b59a-35e5a3bbbfaa)
3. 'Order filled' is displayed for a 'Signing' state
![signing filled](https://github.com/user-attachments/assets/8db529af-0e77-435e-aa66-04c41ddff6b2)
4. (nitpick) I've found an order where a percentage looks glued to the Status column 
![nitpick](https://github.com/user-attachments/assets/e2993b2f-40db-43c9-95c2-26abcbf96d31)
5. "Order cancelled" is not displayed at "fills at" column for a partially filled limit order 
![canelled](https://github.com/user-attachments/assets/afac2ca9-757a-4bd5-88f3-493ba96f25ed)

Thanks!

shoom3301 commented on apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/OrderRow/index.tsx:
```diff
@@ -203,43 +203,83 @@ export function OrderRow({
-    <>
-      {getIsFinalizedOrder(order) ? (
-        order.executionData.partiallyFilled || order.status === OrderStatus.FULFILLED ? (
+  const renderFillsAt = () => {
```
It must be a separate component

shoom3301 replied:
The whole component needs to be decomposed into small pure components.
Please, try to keep component files up to ~200 lines of code.

shoom3301 reviewed (changes requested):
The whole `OrderRow` component needs to be decomposed into small pure components.
Please, try to keep component files up to ~200 lines of code.

anxolin commented:
>The whole OrderRow component needs to be decomposed into small pure components.
> Please, try to keep component files up to ~200 lines of code.

This is the problem of prototype mode is we end up with this situation. If we need, let's take a moment to cleanup now (if it doesn't impact the date) or right after to keep the tech debt in line

shoom3301 commented:
> > The whole OrderRow component needs to be decomposed into small pure components.
> > Please, try to keep component files up to ~200 lines of code.
> 
> This is the problem of prototype mode is we end up with this situation. If we need, let's take a moment to cleanup now (if it doesn't impact the date) or right after to keep the tech debt in line

@anxolin yes, I just realised that.
I've addressed the most of issues [here](https://github.com/cowprotocol/cowswap/pull/5315) and will refactor `OrderRow` once we have all PRs merged in order to avoid merge conflicts.

fairlighteth commented:
> Hey @fairlighteth , great, but there are still some issues;
> 
>     1. Partially filled TWAP parent order does not show this message in the Fills at column
>        ![st](https://private-user-images.githubusercontent.com/70885163/404673998-0a7064d2-002a-41c5-a731-2e0104c5a9d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc0NjIyMzgsIm5iZiI6MTczNzQ2MTkzOCwicGF0aCI6Ii83MDg4NTE2My80MDQ2NzM5OTgtMGE3MDY0ZDItMDAyYS00MWM1LWE3MzEtMmUwMTA0YzVhOWQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIxVDEyMTg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhZDBlNGYyZGVjNzI2MDQ5NTZkYjJiODY5N2U3YjAyODhhZmNiN2Q0YmQ4ZDJhNWE4OTQ1YTNmNjM5MzBiNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FcfSLNOJNxwBlk845E0y5b0ne2T-9HkFmtqW6Av7UFU)
> 
>     2. No status is displayed for a parent TWAP order when it is cancelled+expired:
>        ![image](https://private-user-images.githubusercontent.com/70885163/404674022-f64dc698-8e17-46b5-b59a-35e5a3bbbfaa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc0NjIyMzgsIm5iZiI6MTczNzQ2MTkzOCwicGF0aCI6Ii83MDg4NTE2My80MDQ2NzQwMjItZjY0ZGM2OTgtOGUxNy00NmI1LWI1OWEtMzVlNWEzYmJiZmFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIxVDEyMTg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzOWRmYzkyOTE5YjZlOGQyZjdjNTQ1YjE5ZTExNTMzMjgxZDBjODBkNTc3YmY2YzkxYmJkZjgwYWNhZWZkYmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.kzLM4tw3cGx3wbaxkVytxW58YOIPSoF9joTtJbp7n-o)
> 
>     3. ~'Order filled' is displayed for a 'Signing' state~ Fixed in [Enhance iOS UI, Unify Styling, and Disable Global USD Mode #5309](https://github.com/cowprotocol/cowswap/pull/5309)
>        ![signing filled](https://private-user-images.githubusercontent.com/70885163/404674061-8db529af-0e77-435e-aa66-04c41ddff6b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc0NjIyMzgsIm5iZiI6MTczNzQ2MTkzOCwicGF0aCI6Ii83MDg4NTE2My80MDQ2NzQwNjEtOGRiNTI5YWYtMGU3Ny00MzVlLWFhNjYtMDRjNDFkZGZmNmIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIxVDEyMTg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUyMWNjODliNDA5NGEzNDkzYzYwNzg1MzY0ZjljODUwZDNlOTE0NGI2YmYyNWQ1NWYwMGM4MjdmYjYxNTUwZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qBhkjUenKJhkmc9_r9IG5kFUBjgEy70yZAUDscOZ59Q)
> 
>     4. ~(nitpick) I've found an order where a percentage looks glued to the Status column~ Fixed in [Enhance iOS UI, Unify Styling, and Disable Global USD Mode #5309](https://github.com/cowprotocol/cowswap/pull/5309)
>        ![nitpick](https://private-user-images.githubusercontent.com/70885163/404674144-e2993b2f-40db-43c9-95c2-26abcbf96d31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc0NjIyMzgsIm5iZiI6MTczNzQ2MTkzOCwicGF0aCI6Ii83MDg4NTE2My80MDQ2NzQxNDQtZTI5OTNiMmYtNDBkYi00M2M5LTk1YzItMjZhYmNiZjk2ZDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIxVDEyMTg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4MzQzZjAxYTYwZTU4YzJkNzc4ZGVhNjI3YjUyZDJmNmY1OGNiYTgyNjQwNTE3MmFmYTNjOWIwOTQ4NDQzZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZMxWXqdkuwdPktuvNsb9E86kQj9cH1mSqCnoDeMwpFk)
> 
>     5. "Order cancelled" is not displayed at "fills at" column for a partially filled limit order
>        ![canelled](https://private-user-images.githubusercontent.com/70885163/404674234-afac2ca9-757a-4bd5-88f3-493ba96f25ed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc0NjIyMzgsIm5iZiI6MTczNzQ2MTkzOCwicGF0aCI6Ii83MDg4NTE2My80MDQ2NzQyMzQtYWZhYzJjYTktNzU3YS00YmQ1LTg4ZjMtNDkzYmE5NmYyNWVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTIxVDEyMTg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxOTNiZDEyMzg2N2FiODJkNjQzZTg4NmY3MTgwOWUwOTg5ODFiYmEwZWZiYWMxZWQ0YWI1OTM3NzZhNDVhMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Kb3PQdrNFlRqPYrW-SXZt0ck3gfe1TwGpH0-_hbgamY)
> 
> 
> Thanks!

Addressed comments in https://github.com/cowprotocol/cowswap/pull/5320
